src/calories.rs

```diff
@@ -0,0 +1,92 @@
+#[derive(Debug, PartialEq)]
+pub struct CalorieCounter {
+    third_max: usize,
+    second_max: usize,
+    pub max: usize,
+    current_counter: usize,
+}
+
+impl CalorieCounter {
+    pub fn new(
+        third_max: usize,
+        second_max: usize,
+        max: usize,
+        current_counter: usize,
+    ) -> CalorieCounter {
+        CalorieCounter {
+            third_max,
+            second_max,
+            max,
+            current_counter,
+        }
+    }
+
+    pub fn reset() -> CalorieCounter {
+        CalorieCounter::new(0, 0, 0, 0)
+    }
+
+    pub fn with_next(&self, input: &str) -> CalorieCounter {
+        if input.is_empty() {
+            CalorieCounter::new(self.third_max, self.second_max, self.max, 0)
```

Comment:
I feel like this might actually be a better fit for the struct update syntax - https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax

So you could here do:

```rust
CalorieCounter {  current_counter: 0, ..self }
```
And the same in the other branches,  Less boilerplate, and makes it much more obvious which field is changing each time